coc/http.py

```diff
@@ -415,64 +415,64 @@ def get_player_labels(self, **kwargs):
     async def initialise_keys(self):
         LOG.debug("Initialising keys from the developer site.")
         self.initialising_keys.clear()
-        session = aiohttp.ClientSession()
 
-        body = {"email": self.email, "password": self.password}
-        resp = await session.post("https://developer.clashofclans.com/api/login", json=body)
-        if resp.status == 403:
-            raise InvalidCredentials(resp)
-
-        LOG.info("Successfully logged into the developer site.")
-
-        resp_paylaod = await resp.json()
-        ip = json_loads(base64_b64decode(resp_paylaod["temporaryAPIToken"].split(".")[1] + "====").decode("utf-8"))["limits"][1]["cidrs"][0].split("/")[0]
-
-        LOG.info("Found IP address to be %s", ip)
-
-        resp = await session.post("https://developer.clashofclans.com/api/apikey/list")
-        keys = (await resp.json())["keys"]
-        self._keys.extend(key["key"] for key in keys if key["name"] == self.key_names and ip in key["cidrRanges"])
-
-        LOG.info("Retrieved %s valid keys from the developer site.", len(self._keys))
-
-        if len(self._keys) < self.key_count:
-            for key in (k for k in keys if k["name"] == self.key_names and ip not in k["cidrRanges"]):
-                LOG.info(
-                        "Deleting key with the name %s and IP %s (not matching our current IP address).",
-                        self.key_names, key["cidrRanges"],
+        # Use context manager to automatically clean up after ourselves
+        async with aiohttp.ClientSession() as session:
```

Comment:
If there's a better way of writing all the following code please do, it's pretty ugly. I've redone it a few times and never been happy haha